### DISCUSSION TITLE
Adding a css class for latex-like theorems

### DISCUSSION THREAD

**Discussion by fivoskal** [marp-team/marp#228] [Q&A] [answered]
Hello, 
I would like to incorporate the following classes in my slides:
```
.theorem {
display: block;
font-style: italic;
}
.theorem:before {
content: "Theorem. ";
font-weight: bold;
font-style: normal;
}
.theorem[text]:before {
content: "Theorem (" attr(text) ") ";
}
```
I prepare my slides with the vscode extension but I do not know how to go about it.

Thanks in advance :)

**Comment by yhatt** [comment, marked as answer]
Just wrap CSS in [`<style>` tag.](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style)

**Reply by fivoskal** [reply]
Thanks for the quick response!. I already tried this:
```
<style>
.theorem {
...
```
But it did not work.

**Reply by yhatt** [reply]
Of course just adding style for `.theorem` is not working. You should add the target element such as `<div class="theorem">`. To enable HTML tags, you have to turn on `Markdown > Marp: Enable HTML` from VS Code setting.

**Reply by fivoskal** [reply]
what would a minimum working example look like, something like this? (excuse me but I have no clue about .css whatsoever.)
```
theme: gaia
_class: lead
paginate: true
backgroundColor: #000

<style>
    .theorem {
    display: block;
    font-style: italic;
    }
    .theorem:before {
    content: "Theorem. ";
    font-weight: bold;
    font-style: normal;
    }
    .theorem[text]:before {
    content: "Theorem (" attr(text) ") ";
    }
</style>
---
# A theorem 

<div class="theorem" text="Foo">
Bar
</div>

---

```

**Reply by yhatt** [reply]
```markdown
---
marp: true
---

<style>
  .theorem {
    display: block;
    font-style: italic;
  }
  .theorem:before {
    content: "Theorem. ";
    font-weight: bold;
    font-style: normal;
  }
  .theorem[data-text]:before {
    content: "Theorem (" attr(data-text) ") ";
  }
</style>

# A theorem 

<div class="theorem" data-text="Foo">
Bar
</div>
```

![スクリーンショット 2021-12-07 10 06 06](https://user-images.githubusercontent.com/3993388/144947212-2f2d7fd2-2eb6-4834-831b-608521140f90.png)

**Reply by fivoskal** [reply]
thank you very much!. is there a way I can set the class with a global scope? because it seems it only works only when `<style>` and `<div class="theorem">` are inside the same slide, i.e. `--- ... ---`.

Thank you again for your time! this is a very nice project! I found about it today and I already like it a lot.

**Reply by yhatt** [reply]
No worries. `<style>` is always global in the whole of Markdown document. It is natural behavior as same as HTML. (If needed to scope the style, Marp requires adding [`scoped` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style#attr-scoped) to `<style>`)

**Reply by fivoskal** [reply]
One last thing. Math is not parsed when I use $...$ inside the `<div class="theorem">`. What should I add to the definition of the class for this to happen?

EDIT:
turns out you have to skip a line between `<div>` and `$...$` in order for the last one to be rendered.

**Reply by yhatt** [reply]
:+1: :arrow_right: #152
